platform_channels/lib/src/platform_image_demo.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:typed_data';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:platform_channels/src/image_basic_message_channel.dart';
+
+/// Demonstrates how to use [BasicMessageChannel] to get an image from platform.
+///
+/// The widget uses [Image.memory] to display the image obtained from the
+/// platform.
+class PlatformImageDemo extends StatefulWidget {
+  @override
+  _PlatformImageDemoState createState() => _PlatformImageDemoState();
+}
+
+class _PlatformImageDemoState extends State<PlatformImageDemo> {
+  Uint8List imageData;
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Platform Image Demo'),
+      ),
+      body: Builder(
+        builder: (context) {
+          return Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                Container(
+                  height: size.height / 2,
+                  width: size.width,
+                  child: imageData == null
+                      ? Placeholder()
+                      : Image.memory(
+                          imageData,
+                          fit: BoxFit.fill,
+                        ),
+                ),
+                SizedBox(
+                  height: 16,
+                ),
+                RaisedButton(
+                  onPressed: imageData != null
+                      ? null
+                      : () async {
+                          try {
+                            final result = await PlatformImage.getImage();
+                            setState(() => imageData = result);
+                          } catch (error) {
+                            Scaffold.of(context).showSnackBar(
+                              SnackBar(
+                                content: Text(error.message.toString()),
+                              ),
+                            );
+                          }
+                        },
```

Comment:
While I'm sure this code works, there's a slightly different pattern more commonly used on our team:

1) Change `imageData` to be `Future<Uint8List>` and initialize it to null.
2) In the build method, if `imageData` is null, build the PlaceHolder as you are now.
3) If it's not null, use a FutureBuilder to display the state of the future (uncompleted, completed, error).

This onPressed method would then just need to be `setState(() => imageData = PlatformImage.getImage());`, and the SDK would do the work of waiting on it to complete for you.